packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewGroupDrawingOrderHelper.java

```diff
@@ -65,6 +65,11 @@ public boolean shouldEnableCustomDrawingOrder() {
    * ViewGroup#getChildDrawingOrder}.
    */
   public int getChildDrawingOrder(int childCount, int index) {
+    if (mDrawingOrderIndices != null
+      && (index >= mDrawingOrderIndices.length || mDrawingOrderIndices[index] >= childCount)) {
```

Comment:
Is this ever expected to happen under normal conditions? Can we log a warning here that `mDrawingOrderIndices` is out-of-sync?

Reply:
It will not happen under normal circumstances, but it is not ruled out that some third-party components may add or remove subview in the working thread.